costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -582,7 +582,15 @@ bool Costmap2DROS::getRobotPose(geometry_msgs::PoseStamped& global_pose) const
   // get the global pose of the robot
   try
   {
-    tf_.transform(robot_pose, global_pose, global_frame_);
+    // use current time if possible (makes sure it's not in the future)
+    if (tf_.canTransform(global_frame_, robot_base_frame_, current_time)) {
```

Comment:
I know styling is inconsistent through navigation repo - but let's keep this file as consistent as possible, please move the opening brace to the it's own line.